homeassistant/components/zamg/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config Flow for zamg the Austrian "Zentralanstalt für Meteorologie und Geodynamik" integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_STATION_ID, DOMAIN, LOGGER
+from .sensor import ZamgData, closest_station
+
+
+class ZamgConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for zamg integration."""
+
+    VERSION = 1
+
+    _client: ZamgData
+
+    def _show_setup_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, Any] | None = None,
+        station_id: str = "",
+    ):
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_STATION_ID,
+                        default=user_input.get(CONF_STATION_ID, station_id),
+                    ): str
```

Comment:
If a user didn't set any location it seems to default to Bregenz (11101). The config flow only shows "11101" then without any additional information what this number means. It would be nice to have some kind of dropdown menu or even multiple steps (Bundesland -> Station) or at least some kind of string information of the station name the id belongs to (placeholder). 
If a selection by name is not possible maybe add some information on how to get the station id.

Reply:
If a user did not set a location it is taken Bregenz because its the nearest station to Amsterdam. Currently i am adding a drop down menu to get a list of all available stations.